### PR TITLE
chore: Add Windows Server 2022 Support

### PR DESCRIPTION
**Pull request by jterry75**
Adds Windows Server 2022 Support for OS Platform.

Signed-off-by: Justin Terry <jlterry@amazon.com>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: site/content/docs/manifest/scheduled-job.en.md [hunk 1]
```diff
@@ -120,6 +120,21 @@ platform:
   osfamily: windows_server_2019_full
   architecture: x86_64
 ```
+```yaml
+platform:
+  osfamily: windows_server_2019_full
```
Comment (rawahars) [inline site/content/docs/manifest/scheduled-job.en.md]: Should this be `windows_server_2019_core`?

File: site/content/docs/include/platform.ja.md [hunk 2]
```diff
@@ -18,3 +18,18 @@ platform:
   osfamily: windows_server_2019_full
   architecture: x86_64
 ```
+```yaml
+platform:
+  osfamily: windows_server_2019_full
```
Comment (rawahars) [inline site/content/docs/include/platform.ja.md]: Should this be `windows_server_2019_core`?

**Review by rawahars — APPROVED** [review]

File: site/content/docs/manifest/scheduled-job.en.md (hunk 1 above)
Comment (jterry75) [reply site/content/docs/manifest/scheduled-job.en.md]: Good catch.

File: site/content/docs/include/platform.ja.md (hunk 2 above)
Comment (jterry75) [reply site/content/docs/include/platform.ja.md]: Good catch.

File: site/content/docs/commands/task-run.ja.md [hunk 3]
```diff
@@ -110,3 +110,8 @@ Windows タスクを最小のCPUとメモリで実行します。
 ```console
 $ copilot task run --platform-os WINDOWS_SERVER_2019_CORE --platform-arch X86_64 --cpu 1024 --memory 2048
 ```
+
+Windows 2022 タスクを最小のCPUとメモリで実行します。
```
Comment (hkford) [inline site/content/docs/commands/task-run.ja.md]: I think it is easy to read if there are spaces between English words and Japanese words like "最小の CPU とメモリ"

**Review by hkford — COMMENTED** [review]
Thank you for opening pull request!
And thank you for fixing Japanese document as well.
Left a minor comment.

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go [hunk 4]
```diff
@@ -1514,16 +1514,52 @@ func Test_convertPlatform(t *testing.T) {
 				},
 			},
 			out: template.RuntimePlatformOpts{
-				OS:   template.OSWindowsServerFull,
+				OS:   template.OSWindowsServer2019Full,
 				Arch: template.ArchX86,
 			},
 		},
-		"should return windows server core and x86_64 when platform is 'windows/x86_64'": {
+		"should return windows server 2019 core and x86_64 when advanced config specifies core": {
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/transformers_test.go]: nit:
```suggestion
		"should return windows server 2019 core and x86_64 when advanced config specifies 2019 core": {
```

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go [hunk 5]
```diff
@@ -1514,16 +1514,52 @@ func Test_convertPlatform(t *testing.T) {
...
+					OSFamily: aws.String(manifest.OSWindowsServer2019Core),
+					Arch:     aws.String(manifest.ArchX86),
+				},
+			},
+			out: template.RuntimePlatformOpts{
+				OS:   template.OSWindowsServer2019Core,
+				Arch: template.ArchX86,
+			},
+		},
+		"should return windows server 2022 full and x86_64 when advanced config specifies full": {
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/transformers_test.go]: nit:
```suggestion
		"should return windows server 2022 full and x86_64 when advanced config specifies 2022 full": {
```

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go [hunk 6]
```diff
@@ -1514,16 +1514,52 @@ func Test_convertPlatform(t *testing.T) {
...
+					OSFamily: aws.String(manifest.OSWindowsServer2022Full),
+					Arch:     aws.String(manifest.ArchX86),
+				},
+			},
+			out: template.RuntimePlatformOpts{
+				OS:   template.OSWindowsServer2022Full,
+				Arch: template.ArchX86,
+			},
+		},
+		"should return windows server 2022 core and x86_64 when advanced config specifies core": {
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/transformers_test.go]: nit:
```suggestion
		"should return windows server 2022 core and x86_64 when advanced config specifies 2022 core": {
```

Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/transformers_test.go]: Can we add one more unit test here for 2019 full? 🥺

**Comment by huanjani** [comment]
Wow, thank you so much for this thorough PR!

File: site/content/docs/commands/task-run.ja.md (hunk 3 above)
Comment (jterry75) [reply site/content/docs/commands/task-run.ja.md]: Done please let me know if that is what you wanted.

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go (hunk 4 above)
Comment (jterry75) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: Done good catch.

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go (hunk 5 above)
Comment (jterry75) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: Done good catch.

File: internal/pkg/deploy/cloudformation/stack/transformers_test.go (hunk 6 above)
Comment (jterry75) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: Done good catch.

Comment (jterry75) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: I do have all 4 variants for 2019 Full/2019 Core/2022 Full/2022 Core. Did I miss something? :)

Comment (huanjani) [reply internal/pkg/deploy/cloudformation/stack/transformers_test.go]: My bad! 🤦🏼

**Review by huanjani — APPROVED** [review]
LGTM!

**Review by hkford — APPROVED** [review]
LGTM!Thank you for your contribution!

**Pull request merged**